src/ir_interpreter/interpret_ir.ml

```diff
@@ -369,10 +369,10 @@ and interpret_exp_mut env exp (k : V.value V.cont) =
       | BreakPrim id, [v1] -> find id env.labs v1
       | RetPrim, [v1] -> Option.get env.rets v1
       | ThrowPrim, [v1] -> Option.get env.throws v1
-      | AwaitPrim Type.Fut, [v1] ->
+      | AwaitPrim (AwaitFut _), [v1] ->
         assert env.flavor.has_await;
         await env exp.at (V.as_async v1) k (Option.get env.throws)
-      | AwaitPrim Type.Cmp, [v1] ->
+      | AwaitPrim AwaitCmp, [v1] ->
         assert env.flavor.has_await;
```

Comment:
Won't this still crash when the argument is a Future (see V.as_comp)?

Reply:
No, it won't crash, because desugaring maps the `await* async 42` construct to `AwaitFut true`. So `AwaitCmp` always deals with a computation.